### DISCUSSION TITLE
[Github Actions] Contexts and steps if conditionals are unusable with incomplete docs

### DISCUSSION THREAD

**Discussion by geekodour** [community/community#26119] [Actions]
<p>When I echo the context for <code>steps</code> as mentioned in the docs i get <code>{}</code> and <code>job</code> just has two fields. </p>
<p>GitHub Actions Contexts:</p>
<p><a href="https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#contexts" rel="nofollow noopener">https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#contexts</a></p>
<p>Status check functions:</p>
<p><a href="https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#status-check-functions" rel="nofollow noopener">https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#status-check-functions</a></p>
<p>In the workflow syntax it is clear that <code>failure</code> should have been <code>&lt;font face="terminal,monaco"&gt;failure()&lt;/font&gt;</code></p>
<p><a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#example-using-status-check-functions" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#example-using-status-check-functions</a></p>
<p>Currently selecting on which <code>step</code>'s success/failure should another step be triggered is not working it seems.</p>
<p>i.e</p>
<pre><code>steps:
name: my first step
if: job.steps.&lt;step id&gt;.status == failure
</code></pre>
<p>^ the above snippet from the doc is not working.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/geekodour">@geekodour</a>,</p>
<p>Thanks again for taking the time to report this, we are working on it internally and I’ll post an update here as soon as I have news.</p>
<p>Thanks again. Meanwhile, we’re looking forward to seeing you around!</p>

**Comment by geekodour** [comment]
<p>Just to be even more clear about my question, I think <code>job.steps.\&lt;step_id\&gt;.status</code> does not exist yet. as job context is just</p>
<pre><code>{
  "status": "Success"
}
</code></pre>
<p>and steps context is just {}</p>

**Comment by msuzoagu** [comment]
<p><a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> any update on this?  </p>
<p>While i understand that the product is still in beta, I think it is more useful to only release docs for api/endpoints that work; not things that we hope will work in the future.</p>
<p>Time is my most valuable asset and it is really frustrating to invest time using/learning a new product only to learn that particular endpoint isn’t working “YET” and no clear instructions on alternatives to use in the meantime.</p>

**Comment by hamelsmu** [comment]
<p>Try</p>
<pre><code>if: job.steps.&lt;step id&gt;.status == failure()
</code></pre>

**Comment by Herohtar** [comment]
<p>It doesn’t work either way because <strong>job.steps.&lt;step id&gt;.status</strong> is null. <strong>job</strong> is the following simple object:</p>
<pre><code>{
    "status": "Success"
}
</code></pre>
<p>As you can see, it doesn’t even have a <strong>steps</strong> property, let alone anything below that.</p>

**Comment by meeDamian** [comment]
<p>Since I’m finding many parts of the docs are still more like a whishlist, I’m starting this repo, that tracks progress of the missing features: <a href="https://github.com/meeDamian/github-actions-bugs/actions" rel="nofollow noopener">https://github.com/meeDamian/github-actions-bugs/actions</a>, I’ll be adding one feature check, per scheduled daily workflow file.  Feel free to contribute, or peek-in to see if it’s already implemented :).</p>

**Comment by chrispat** [comment, marked as answer]
<p>That is a bug in the docs that I have opened a PR for.  You can’t compare the status of an individual step, you can just check the status of the entire job to that point.  The built in status functions in that doc compare against the overall job status.</p>
